packages/compiler-core/src/transforms/transformElement.ts

```diff
@@ -597,6 +597,7 @@ export function buildProps(
       }
 
       if (isVBind && isStaticArgOf(arg, 'ref') && context.scopes.vFor > 0) {
+        hasRef = true
```

Comment:
The tests all seem to pass even if this line is omitted. Perhaps there should be another test case added to cover this scenario too?

Reply:
Here is to address the issue at https://github.com/vuejs/core/issues/9239#issuecomment-1727023890, the function cannot be resolved in compile so I can't add a unit test for it.